File: TrialTest/ChapterThreePython/validating_input.py
num_student = int(input("Enter the number of students: "))

number_of_failures = 0
number_of_passes = 0
for i in range(num_student):
    validate = int(input("Enter a number '1' for pass '2' for fail : "))
    while(validate != 1 and validate != 2):
        validate = int(input("invalid input. Enter a number '1' for pass '2' for fail : "))
    if(validate == 1):
        number_of_passes += 1

    elif(validate == 2):
        number_of_failures += 1
print(f"{number_of_failures}  are the number of failures")
print(f"{number_of_passes}  are the number of passes")

File: TrialTest/ChapterFourJava/ChapterFourExercise/src/ValidatingInput.java
import java.util.Scanner;

public class ValidatingInput {
    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);


        System.out.print("Please enter a number of students: ");
        int students = input.nextInt();
        int failures = 0;
        int passes = 0;
        for (int num = 1; num <= students; num++) {
            System.out.print("Please enter a student grade '1' for pass '2' for fail: ");
            int grade = input.nextInt();
            while (grade != 1 && grade != 2) {
                System.out.print("Enter a grade between 1 and 2: ");
                grade = input.nextInt();
            }
                if (grade == 1) {
                    passes += 1;
                } else if (grade == 2) {
                    failures += 1;
                }
        }


            System.out.println("The number of student passes is: " + passes);
            System.out.println("The number of student failures is: " + failures);



    }
}
